packages/astro/src/content/loaders/glob.ts

```diff
@@ -63,41 +64,75 @@ export function glob(globOptions: GlobOptions): Loader {
 
 	const fileToIdMap = new Map<string, string>();
 
-	async function syncData(
-		entry: string,
-		base: URL,
-		{ logger, parseData, store }: LoaderContext,
-		entryType?: ContentEntryType
-	) {
-		if (!entryType) {
-			logger.warn(`No entry type found for ${entry}`);
-			return;
-		}
-		const fileUrl = new URL(entry, base);
-		const { body, data } = await entryType.getEntryInfo({
-			contents: await fs.readFile(fileUrl, 'utf-8'),
-			fileUrl,
-		});
-
-		const id = generateId({ entry, base, data });
-
-		const filePath = fileURLToPath(fileUrl);
-
-		const parsedData = await parseData({
-			id,
-			data,
-			filePath,
-		});
-
-		store.set(id, parsedData, body, filePath);
-
-		fileToIdMap.set(filePath, id);
-	}
-
 	return {
 		name: 'glob-loader',
 		load: async (options) => {
-			const { settings, logger, watcher } = options;
+			const { settings, logger, watcher, parseData, store } = options;
+			const { h64ToString } = await xxhash();
```

Comment:
It might be worth providing a `generateDigest` utility, based on this.

Reply:
Unfamiliar with xxhash. Is that preferred over web crypto or node crypto for cross-platform compat?

Reply:
It's a non-cryptographic hash, which is orders of magnitude faster than crypto hash functions. It's designed specifically for this kind of case. https://github.com/sebastian-software/node-hash-comparison